odoo/models.py

```diff
@@ -4477,6 +4481,7 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
         by specifying ``access_rights_uid=1`` to bypass access rights check, but not ir.rules!
         This is ok at the security level because this method is private and not callable through XML-RPC.
 
+        :param order: optional specific ordering. None means default (=model._order). False means not ordering
```

Comment:
Why's this part of the same PR?

Also given `order` is `str | None`, might make more sense to have this occur on `order=''` instead? Explicitly providing an empty order is more logical than adding a boolean placeholder.

It would require a change to `_generate_order_by` to handle `order_spec is None` specifically but I think that's a smaller and simpler change, and might benefit other cases than just search.

Can't help but think we've had issues / PRs on that subject in the past.

Reply:
I agree with @xmo-odoo 's remark.  I once saw code where a developer used an explicit `order=False` in `search()` to mean... something, and that something currently behaves as `order=None`.  Using `order=""` makes more sense to me.

Reply:
If you remember, this was also discussed for #46392. It kind of sucks to change the public API of `search()` to something that feels awkward, just for the purpose of shaving a few % of query time in very specific cases.

On the other hand, we can make some specific changes to the internals of `expression.py` that may yield similar benefits, without changing any API. This is more akin to what we did in #52368.

@KangOl was proposing something similar:
```patch
diff --git odoo/osv/expression.py odoo/osv/expression.py
index ab4b546fa16..fc4e0d9fbce 100644
--- odoo/osv/expression.py
+++ odoo/osv/expression.py
@@ -680,6 +680,7 @@ class expression(object):
                 # use a subquery bypassing access rules and business logic
                 domain = [(path[1], operator, right)] + field.get_domain_list(model)
                 query = comodel.with_context(**field.context)._where_calc(domain)
+                query.order = None
                 subquery, subparams = query.select('"%s"."%s"' % (comodel._table, field.inverse_name))
                 push(('id', 'inselect', (subquery, subparams)), model, alias, internal=True)
 
@@ -749,6 +750,7 @@ class expression(object):
 
                     if isinstance(ids2, Query) and comodel._fields[field.inverse_name].store:
                         op1 = 'not inselect' if operator in NEGATIVE_TERM_OPERATORS else 'inselect'
+                        ids2.order = None
                         subquery, subparams = ids2.select('"%s"."%s"' % (comodel._table, field.inverse_name))
                         push(('id', op1, (subquery, subparams)), model, alias, internal=True)
                     elif ids2 and comodel._fields[field.inverse_name].store:
@@ -812,6 +814,7 @@ class expression(object):
                     if isinstance(ids2, Query):
                         # rewrite condition in terms of ids2
                         subop = 'not inselect' if operator in NEGATIVE_TERM_OPERATORS else 'inselect'
+                        ids2.order = None
                         subquery, subparams = ids2.select()
                         query = 'SELECT "%s" FROM "%s" WHERE "%s" IN (%s)' % (rel_id1, rel_table, rel_id2, subquery)
                         push(('id', subop, (query, subparams)), model, alias, internal=True)
```